src/ansys/units/dimensions.py

```diff
@@ -19,13 +19,20 @@ class Dimensions:
     ----------
     dimensions : dict, optional
         Dictionary of {``BaseDimensions``: exponent, ...}.
+    copy_from : Dimensions, optional
+        A previous instance of Dimensions.
     """
 
     def __init__(
         self,
         dimensions: dict[ansunits.BaseDimensions, Union[int, float]] = None,
+        copy_from: ansunits.Dimensions = None,
     ):
         dimensions = dimensions or {}
+
+        if copy_from:
+            dimensions = {**copy_from._dimensions, **dimensions}
+
         self._dimensions = dimensions.copy()
         for x, y in dimensions.items():
             if not isinstance(x, ansunits.BaseDimensions):
```

Comment:
With the addition new `copy_from` argument, this whole chunk of code deserves a bit of refactoring, because the two dicts can be handled in a general way (also avoiding extra copies in some cases). 

Note that there is an implicit precedence here (the second dict wins in line 34!).  This is not commented or documented, or is it? Is this intended by the way.

Also, the loop is over the local dict - is that intended? I guess so.

Reply:
My goal here was to be able to add dimensions to a existing dimensions instance to create a new dimension. 
```
import ansys.units as ansunits

dims = ansunits.BaseDimensions
dim_1 = ansunits.Dimensions({dims.MASS: 2, dims.LENGTH: 1})
dim_1 #{'MASS': 2, 'LENGTH': 1}
dict1 = {dims.MASS: 1, dims.LIGHT: 1}
dims_2 = ansunits.Dimensions(dimensions=dict1, copy_from=dim_1)
dims_2 #{'MASS': 1, 'LENGTH': 1, 'LIGHT': 1}
```
What would be expected if there was both a `copy_from` and `dimensions` arguments?

Reply:
I think it's fine as is, just wanted to make sure it was doing what you intended, but also that we are properly documenting the behaviour.

Reply:
But regarding the refactoring I mentioned:
```
# n.b. dimensions here is just the passed-in dict, nothing else needs to be checked here
if dimensions and not all(isinstance(k, ansunits.BaseDimensions) for k in dimensions):
  raise ...
# now just make the copy from anything that's passed in
self._dimensions = {**(copy_from._dimensions or {}), **(dimensions or {})}
```

Reply:
I have made some changes here:
```
 def __init__(
        self,
        dimensions: dict[ansunits.BaseDimensions, Union[int, float]] = None,
        copy_from: ansunits.Dimensions = None,
    ):
        dimensions = dimensions or {}
        self._dimensions = {
            **(copy_from._dimensions if copy_from else {}),
            **(dimensions),
        }

        for x, y in dimensions.items():
            if not isinstance(x, ansunits.BaseDimensions):
                raise DimensionsError.INCORRECT_DIMENSIONS()
            if y == 0:
                del self._dimensions[x]
```
I would like to keep the for loop, I believe removing dimensions of 0 exponent is required.